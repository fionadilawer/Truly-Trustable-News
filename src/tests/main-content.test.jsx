import { render, screen } from "@testing-library/react";
import MainContent from "../component/main-content";
import { describe, it, expect } from 'vitest';

describe('main-news-component', () => {
  it('renders the news-cards', () => {
    const { container } = render(<MainContent />);
    expect(container).toMatchSnapshot();
  });

  it('changes the page route after the news-card is clicked', () => {
    render(<MainContent />);

    const newscard = screen.getByRole('link');

    expect(newscard).toHaveAttribute('href', 'news1')
  });
});
