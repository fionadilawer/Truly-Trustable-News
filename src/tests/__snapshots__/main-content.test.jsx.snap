// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`main-news-component > renders the news-cards 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="content"
      style="grid-template-columns: 1fr 1fr;"
    >
      <a
        class="item"
        href="news1"
      >
        <div
          class="itemwrap1"
        >
          <img
            id="img"
            src="./src/assets/images/poco.jpg"
          />
          <h1>
            this is the heading
          </h1>
        </div>
        <p>
          bla bla bla bla bla bla bla lb narendra modi bla lbal dlad hmklajmkclamrk nlknakll mfdsklm fkdsmnf kldjf kldsngjdf jgkldn kgj ksngkdfm gfdkng kdfmgdklm gkldfg kldfjgklrf kgldfnvkl dfkglv kdfog kdfklgjm kdflgm  
        </p>
      </a>
      <div
        class="item"
      >
        <div
          class="itemwrap2"
        >
          <img
            id="img"
            src="./src/assets/images/poco.jpg"
          />
          <h1>
            this is the heading
          </h1>
        </div>
        <p>
          bla bla bla bla bla bla bla lb narendra modi bla lbal dlad hmklajmkclamrk nlknakll mfdsklm fkdsmnf kldjf kldsngjdf jgkldn kgj ksngkdfm gfdkng kdfmgdklm gkldfg kldfjgklrf kgldfnvkl dfkglv kdfog kdfklgjm kdflgm  
        </p>
      </div>
      <div
        class="item"
      >
        <div
          class="itemwrap1"
        >
          <img
            id="img"
            src="./src/assets/images/poco.jpg"
          />
          <h1>
            this is the heading
          </h1>
        </div>
        <p>
          bla bla bla bla bla bla bla lb narendra modi bla lbal dlad hmklajmkclamrk nlknakll mfdsklm fkdsmnf kldjf kldsngjdf jgkldn kgj ksngkdfm gfdkng kdfmgdklm gkldfg kldfjgklrf kgldfnvkl dfkglv kdfog kdfklgjm kdflgm  
        </p>
      </div>
      <div
        class="item"
      >
        <div
          class="itemwrap2"
        >
          <img
            id="img"
            src="./src/assets/images/poco.jpg"
          />
          <h1>
            this is the heading
          </h1>
        </div>
        <p>
          bla bla bla bla bla bla bla lb narendra modi bla lbal dlad hmklajmkclamrk nlknakll mfdsklm fkdsmnf kldjf kldsngjdf jgkldn kgj ksngkdfm gfdkng kdfmgdklm gkldfg kldfjgklrf kgldfnvkl dfkglv kdfog kdfklgjm kdflgm  
        </p>
      </div>
      <div
        class="item"
      >
        <div
          class="itemwrap1"
        >
          <img
            id="img"
            src="./src/assets/images/poco.jpg"
          />
          <h1>
            this is the heading
          </h1>
        </div>
        <p>
          bla bla bla bla bla bla bla lb narendra modi bla lbal dlad hmklajmkclamrk nlknakll mfdsklm fkdsmnf kldjf kldsngjdf jgkldn kgj ksngkdfm gfdkng kdfmgdklm gkldfg kldfjgklrf kgldfnvkl dfkglv kdfog kdfklgjm kdflgm  
        </p>
      </div>
      <div
        class="item"
      >
        <div
          class="itemwrap2"
        >
          <img
            id="img"
            src="./src/assets/images/poco.jpg"
          />
          <h1>
            this is the heading
          </h1>
        </div>
        <p>
          bla bla bla bla bla bla bla lb narendra modi bla lbal dlad hmklajmkclamrk nlknakll mfdsklm fkdsmnf kldjf kldsngjdf jgkldn kgj ksngkdfm gfdkng kdfmgdklm gkldfg kldfjgklrf kgldfnvkl dfkglv kdfog kdfklgjm kdflgm  
        </p>
      </div>
      <div
        class="item"
      >
        <div
          class="itemwrap1"
        >
          <img
            id="img"
            src="./src/assets/images/poco.jpg"
          />
          <h1>
            this is the heading
          </h1>
        </div>
        <p>
          bla bla bla bla bla bla bla lb narendra modi bla lbal dlad hmklajmkclamrk nlknakll mfdsklm fkdsmnf kldjf kldsngjdf jgkldn kgj ksngkdfm gfdkng kdfmgdklm gkldfg kldfjgklrf kgldfnvkl dfkglv kdfog kdfklgjm kdflgm  
        </p>
      </div>
      <div
        class="item"
      >
        <div
          class="itemwrap2"
        >
          <img
            id="img"
            src="./src/assets/images/poco.jpg"
          />
          <h1>
            this is the heading
          </h1>
        </div>
        <p>
          bla bla bla bla bla bla bla lb narendra modi bla lbal dlad hmklajmkclamrk nlknakll mfdsklm fkdsmnf kldjf kldsngjdf jgkldn kgj ksngkdfm gfdkng kdfmgdklm gkldfg kldfjgklrf kgldfnvkl dfkglv kdfog kdfklgjm kdflgm  
        </p>
      </div>
      <div
        class="item"
      >
        <div
          class="itemwrap1"
        >
          <img
            id="img"
            src="./src/assets/images/poco.jpg"
          />
          <h1>
            this is the heading
          </h1>
        </div>
        <p>
          bla bla bla bla bla bla bla lb narendra modi bla lbal dlad hmklajmkclamrk nlknakll mfdsklm fkdsmnf kldjf kldsngjdf jgkldn kgj ksngkdfm gfdkng kdfmgdklm gkldfg kldfjgklrf kgldfnvkl dfkglv kdfog kdfklgjm kdflgm  
        </p>
      </div>
      <div
        class="item"
      >
        <div
          class="itemwrap2"
        >
          <img
            id="img"
            src="./src/assets/images/poco.jpg"
          />
          <h1>
            this is the heading
          </h1>
        </div>
        <p>
          bla bla bla bla bla bla bla lb narendra modi bla lbal dlad hmklajmkclamrk nlknakll mfdsklm fkdsmnf kldjf kldsngjdf jgkldn kgj ksngkdfm gfdkng kdfmgdklm gkldfg kldfjgklrf kgldfnvkl dfkglv kdfog kdfklgjm kdflgm  
        </p>
      </div>
      <div
        class="item"
      >
        <div
          class="itemwrap1"
        >
          <img
            id="img"
            src="./src/assets/images/poco.jpg"
          />
          <h1>
            this is the heading
          </h1>
        </div>
        <p>
          bla bla bla bla bla bla bla lb narendra modi bla lbal dlad hmklajmkclamrk nlknakll mfdsklm fkdsmnf kldjf kldsngjdf jgkldn kgj ksngkdfm gfdkng kdfmgdklm gkldfg kldfjgklrf kgldfnvkl dfkglv kdfog kdfklgjm kdflgm  
        </p>
      </div>
      <div
        class="item"
      >
        <div
          class="itemwrap2"
        >
          <img
            id="img"
            src="./src/assets/images/poco.jpg"
          />
          <h1>
            this is the heading
          </h1>
        </div>
        <p>
          bla bla bla bla bla bla bla lb narendra modi bla lbal dlad hmklajmkclamrk nlknakll mfdsklm fkdsmnf kldjf kldsngjdf jgkldn kgj ksngkdfm gfdkng kdfmgdklm gkldfg kldfjgklrf kgldfnvkl dfkglv kdfog kdfklgjm kdflgm  
        </p>
      </div>
      <div
        class="item"
      >
        <div
          class="itemwrap1"
        >
          <img
            id="img"
            src="./src/assets/images/poco.jpg"
          />
          <h1>
            this is the heading
          </h1>
        </div>
        <p>
          bla bla bla bla bla bla bla lb narendra modi bla lbal dlad hmklajmkclamrk nlknakll mfdsklm fkdsmnf kldjf kldsngjdf jgkldn kgj ksngkdfm gfdkng kdfmgdklm gkldfg kldfjgklrf kgldfnvkl dfkglv kdfog kdfklgjm kdflgm  
        </p>
      </div>
      <div
        class="item"
      >
        <div
          class="itemwrap2"
        >
          <img
            id="img"
            src="./src/assets/images/poco.jpg"
          />
          <h1>
            this is the heading
          </h1>
        </div>
        <p>
          bla bla bla bla bla bla bla lb narendra modi bla lbal dlad hmklajmkclamrk nlknakll mfdsklm fkdsmnf kldjf kldsngjdf jgkldn kgj ksngkdfm gfdkng kdfmgdklm gkldfg kldfjgklrf kgldfnvkl dfkglv kdfog kdfklgjm kdflgm  
        </p>
      </div>
    </div>
    <div
      class="sidebar"
    />
  </div>
</div>
`;
